File: Java/Motherboard.java
import java.util.ArrayList;
import java.util.List;

public class Motherboard extends Komponen 
{
    private String chipset;
    private String socketCpu;
    private List<Ram> ramList;
    
    public Motherboard(String chipset, String socketCpu, List<Ram> ramList, String merk, String nama) {
        super(merk, nama);
        this.chipset = chipset;
        this.socketCpu = socketCpu;
        this.ramList = ramList != null ? ramList : new ArrayList<>();
    }
    
    public Motherboard() {
        super("", "");
        this.chipset = "";
        this.socketCpu = "";
        this.ramList = new ArrayList<>();
    }
    
    public void setChipset(String chipset) {
        this.chipset = chipset;
    }
    
    public void setSocketCpu(String socketCpu) {
        this.socketCpu = socketCpu;
    }
    
    public void setRamList(List<Ram> ramList) {
        this.ramList = ramList;
    }
    
    public String getChipset() {
        return chipset;
    }
    
    public String getSocketCpu() {
        return socketCpu;
    }
    
    public List<Ram> getRamList() {
        return ramList;
    }
    
    public void addRam(Ram ram) {
        this.ramList.add(ram);
    }
}

File: Python/Motherboard.py
from Ram import Ram
from Komponen import Komponen

class Motherboard(Komponen):
    def __init__(self, chipset="", socket_cpu="", ram_list=[], merk="", nama=""):
        super().__init__(merk, nama)
        self.__chipset = chipset
        self.__socket_cpu = socket_cpu
        self.__ram_list = ram_list
    
    def set_chipset(self, chipset):
        self.__chipset = chipset
    
    def set_socketCpu(self, socket_cpu):
        self.__socket_cpu = socket_cpu
    
    def set_ramList(self, ram_list):
        self.__ram_list = ram_list
    
    def get_chipset(self):
        return self.__chipset
    
    def get_socketCpu(self):
        return self.__socket_cpu
    
    def get_ramList(self):
        return self.__ram_list
    
    def addRam(self, ram):
        self.__ram_list.append(ram)